feat: добавил импорт пользователей из Google Spreadsheet

VladBaykin opened the pull request:
Добавляет импорт пользователей из Google Spreadsheet

VladBaykin committed:
feat: добавил импорт пользователей из Google Spreadsheet

DarkRubin commented on src/main/java/com/itmentorcommunityplatform/dataimporter/service/UserImportService.java:
```diff
@@ -0,0 +1,77 @@
+                    log.warn("Invalid telegram id '{}' at row index {}, skipping.", tgRaw, i);
+                    continue;
+                }
+                String role = (props.getAdminIds() != null && props.getAdminIds().contains(tgId)) ? "ADMIN" : "STUDENT";
```
ADMIN и STUDENT явно будут ещё переиспользоватся в будущем, так что стоит сразу создать enum для них

DarkRubin commented on src/main/java/com/itmentorcommunityplatform/dataimporter/controller/UserImportController.java:
```diff
@@ -0,0 +1,37 @@
+
+    @PostMapping("/start-users-import")
+    public ResponseEntity<?> startUsersImport(@RequestHeader(value = "X-User-Roles", required = false)
+                                                  String rolesHeader) {
```
Можно сразу `List<String> roles`

DarkRubin commented on src/main/java/com/itmentorcommunityplatform/dataimporter/service/UserImportService.java:
```diff
@@ -0,0 +1,77 @@
+    private final DataImporterProperties props;
+
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(r -> {
+        Thread t = new Thread(r);
```
`r -> new Thread(r, "users-import-thread")`

VladBaykin committed:
fix: переход на Workload Identity Federation и правки
- Удалён credentials.json, вместо него используется Workload Identity Federation
- Обновлены настройки аутентификации Google Sheets API
- Исправлены мелкие замечания (именование ролей, запуск нового Thread)
- В README добавлена инструкция по локальному запуску и тестированию

VladBaykin committed:
fix: обновил workflow

VladBaykin committed:
fix: обновил workflow, внес изменения в Google Console

VladBaykin committed:
fix: обновил README:
- добавил команды аутентификации
